### PR TITLE
Keep build_from_cache in sync with test/build.yml

### PR DESCRIPTION
**Pull request by kevinemoore**
The test test_build_from_cache contains an assertion that verifies that
the cache is actually filled. The hash key in the cache is generated
from both the source file and kwargs so it needs to be kept in sync
with the build file (test/build.yml).

Fix an indentation bug in test/build.yml.

### CODE REVIEW

File: compiler/quilt/test/build.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@
 contents:
   dataframes:
     csv: 
-    kwargs:
-      parse_dates: ['Date0']
-    file: data/10KRows13Cols.csv
+      kwargs:
```
Comment (dimaryaz) [inline compiler/quilt/test/build.yml]: @aneesh just fixed this one - but the other change is still needed I'm guessing.

Comment (kevinemoore) [reply compiler/quilt/test/build.yml]: This is one way to fix that broken test, but @akarve might want to fix it another way. This PR is targeting his branch, so it's up to him.

Comment (akarve) [reply compiler/quilt/test/build.yml]: I merged those fixes into `master`. Can you catch this branch up?

File: compiler/quilt/test/test_build.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def test_build_from_cache(self):
 
         # Verify cache contents
         srcpath = os.path.join(mydir, 'data/10KRows13Cols.csv')
-        path_hash = build._path_hash(srcpath, 'csv', {})
+        path_hash = build._path_hash(srcpath, 'csv',  {'parse_dates': ['Date0']})
```
Comment (akarve) [inline compiler/quilt/test/test_build.py]: this works but it's super fragile; @kevinemoore can you get this info from build.yml? other unit tests will need the logic.

File: compiler/quilt/test/build.yml (hunk 1 above)
Comment (kevinemoore) [reply compiler/quilt/test/build.yml]: This PR is merging into cacheBug. It's caught up with that branch.

**Pull request merged**